modules/matterhorn-asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerWithSecurity.java

```diff
@@ -73,12 +77,52 @@ public AssetManagerWithSecurity(
     this.secSvc = secSvc;
   }
 
-  @Override public Snapshot takeSnapshot(String owner, MediaPackage mp) {
-    final AccessControlList acl = authSvc.getActiveAcl(mp).getA();
+  private boolean isAuthorizedByAcl(AccessControlList acl, String action) {
     final User user = secSvc.getUser();
     final Organization org = secSvc.getOrganization();
-    if (AccessControlUtil.isAuthorized(acl, user, org, WRITE_ACTION)) {
+    return AccessControlUtil.isAuthorized(acl, user, org, action);
+  }
+
+  private boolean isAuthorizedByAcl(MediaPackage mp, String action) {
+    final AccessControlList acl = authSvc.getActiveAcl(mp).getA();
+    return isAuthorizedByAcl(acl, action);
+  }
+
+  private boolean isAuthorizedByAcl(Version version, String mpId, String action) {
+    final AQueryBuilder query = q();
+    final AResult results = query.select(query.snapshot())
+                              .where(query.mediaPackageId(mpId))
+                              .run();
+
+    if (results.getSize() == 1) {
+      final Opt<Snapshot> snapShot = results.getRecords().head2().getSnapshot();
+      if (snapShot.isSome()) {
+        final MediaPackage mp = snapShot.get().getMediaPackage();
+        String xacmlId = null;
+        for (MediaPackageElement mpe : mp.getElements()) {
+          if (mpe.toString().contains("security-policy-episode")) {
```

Comment:
I' am do not know why you are iterating over all elements, my be it is the only way.. 
But, this seems also be a solution, so you do not need to iterate over all elements ?:
MediapackageElement mpe = mp.getElementById("security-policy-episode");
xacmlId = mpe.getIdentifier();

Reply:
For some reason I couldn't get the episode ACL by id or reference at the time, hence the iteration.
 
Your suggestion helped me in steamlining this method (its so simple now!).